Automattic/woocommerce-payments
# Introduce `wcpay_payment_request_is_cart_supported` filter.

Fixes #7438

#### Changes proposed in this Pull Request





Introduces the filter `wcpay_payment_request_is_cart_supported` as a sibling filter to `wcpay_payment_request_is_product_supported`. 

This is to allow extenders to specify whether a product supports the express payment options based on attributes other than type.

An example is bookings products generally support the express pay options in the cart and at checkout, but for bookings requiring confirmation the express pay options are not supported as payment is not processed at the time.



Reviewers:

* Please advise if and how you would like this to be tested.




#### Testing instructions





1. Enable express payments options (Apple or Google Pay)
   * testing Apple Pay requires you use the Safari browser
   * testing Google Pay requires you use the Chrome browser
2. Create a simple product
3. Add the product to your cart
4. View the cart
5. The express pay button should display
6. Add the mini plugin `add_filter( 'wcpay_payment_request_is_cart_supported', '__return_false' )` to your mu-plugins folder
7. Refresh the cart, the express pay button should not display
8. Delete the mini-plugin to avoid confusion later

-------------------


**Post merge**
